osu.Game/Overlays/Toolbar.cs

```diff
@@ -39,13 +47,39 @@ protected override void PopOut()
             FadeOut(transition_time, EasingTypes.InQuint);
         }
 
-        public override void Load(BaseGame game)
+        private void CheckUser()
         {
-            base.Load(game);
+            if (user == null)
+                return;
+            if (user.UserId == 0)
+            {
+                toolbarUserButton.UpdateButton(1, "Not signed in");
+            }
+            else if (user.UserId != toolbarUserButton.UserId)
+            {
+                toolbarUserButton.UpdateButton(user.UserId, user.Name);
+                toolbarUserButton.UserId = user.UserId;
+            }
+        }
 
-            RelativeSizeAxes = Axes.X;
-            Size = new Vector2(1, height);
+        private void InitUser()
+        {
+            MyUser req = new MyUser();
+            api.Queue(req);
+            req.Success += delegate (User MyUser)
+            {
+                this.user = MyUser;

```

Comment:
you're throwing out the old user and replacing it every time this is called. please no.